Apidaze.SDK.Tests.Integration/IvrExample/Program.cs

```diff
@@ -0,0 +1,335 @@
+        {
+            switch (context.Request.HttpMethod)
+            {
+                case "GET":
+                    response.ContentType = "text/xml";
+
+                    var step1 = ApidazeScript.Build()
+                        .AddNode(Speak.WithText(
+                            "Our text to speech leverages Google's cloud APIs to offer the best possible solution"))
+                        .AddNode(Wait.SetDuration(1))
+                        .AddNode(new Speak()
+                        {
+                            LangEnum = LangEnum.ENGLISH_AUSTRALIA,
+                            Voice = VoiceEnum.MALE_A,
+                            Text = "A wide variety of voices and languages are available.Here are just a few"
+                        }).AddNode(Wait.SetDuration(1))
+                        .AddNode(new Speak() { LangEnum = LangEnum.FRENCH_FRANCE, Text = "Je peux parler français" }).AddNode(
+                            Wait.SetDuration(1)).AddNode(new Speak() { LangEnum = LangEnum.GERMAN, Text = "Auch deutsch" })
+                        .AddNode(Wait.SetDuration(1)).AddNode(new Speak()
+                        {
+                            LangEnum = LangEnum.JAPANESE,
+                            Text = "そして日本人ですら"
+                        }).AddNode(Wait.SetDuration(1)).AddNode(new Speak()
+                        {
+                            Text =
+                                "You can see our documentation for a full list of supported languages and voices for them.  We'll take you back to the menu for now."
+                        }).AddNode(Wait.SetDuration(2))
+                        .ToXml();
+
+                    handled = WriteResponseStream(response, step1);
+                    break;
+            }
+
+            return handled;
+        }
+
+        private static bool GetStep2(HttpListenerContext context, HttpListenerResponse response, bool handled)
+        {
+            switch (context.Request.HttpMethod)
+            {
+                case "GET":
+                    //Get the current settings
+                    response.ContentType = "text/xml";
+
+                    var step2 = ApidazeScript.Build()
+                        .AddNode(Speak.WithText("You will now be joined to an echo line."))
+                        .AddNode(Echo.SetDuration(500))
+                        .ToXml();
+
+                    handled = WriteResponseStream(response, step2);
+                    break;
+            }
+
+            return handled;
+        }
+
+        private static void ReturnExceptionResponse(HttpListenerResponse response, Exception e)
+        {
+            response.StatusCode = 500;
+            response.ContentType = "text/plain";
+            var buffer = Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(e));
+            response.ContentLength64 = buffer.Length;
+            response.OutputStream.Write(buffer, 0, buffer.Length);
```

Comment:
To avoid code duplication, we could extract the whole response header into a private method that will give us the header we need based on some attributes, or a default if nothing is special is needed